src/main/kotlin/ua/marchenko/artauction/common/annotation/scheduled/annotation/CustomScheduled.kt

```diff
@@ -0,0 +1,7 @@
+package ua.marchenko.artauction.common.annotation.scheduled.annotation
+
+import ua.marchenko.artauction.common.annotation.scheduled.scheduledDetails.enums.Day
+
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FUNCTION)
+annotation class CustomScheduled(val day: Day, val hours: Int = 0, val minutes: Int = 0, val seconds: Int = 0)
```

Comment:
you can make day nullable an null will mean 'run on each day' and then you can use embedded enum

Reply:
Good point, but I can't do it due to

> Annotation parameters cannot have nullable types, because the JVM does not support storing null as a value of an annotation attribute.

https://kotlinlang.org/docs/annotations.html#constructors

Reply:
create wrapper class then) which will have nullable enum property)